ocaml/quicktest/qt_filter.ml

```diff
@@ -128,7 +128,24 @@ module SR = struct
 
   let only sr () = [get_sr_info !A.rpc !session_id sr]
 
-  let all_srs = lazy (list_selected_srs !A.rpc !session_id)
+  let all_srs =
+    let is_shared_or_local sr_info =
+      (** allows us to ignore srs which don't belong to us.
+        * this remedies some failing tests if we run the quicktest binary on a host in a pool *)
+      let sr = sr_info.Qt.sr in
+      let session_id = !session_id in
+      let rpc = !A.rpc in
+      let pbds = Client.Client.SR.get_PBDs ~rpc ~session_id ~self:sr in
+      match pbds with
+      | [pbd] -> begin
+          let pbd_host = Client.Client.PBD.get_host ~rpc ~session_id ~self:pbd in
+          let pbd_host_uuid = Client.Client.Host.get_uuid ~rpc ~session_id ~self:pbd_host in
+          pbd_host_uuid = A.localhost_uuid (* sr is not shared, test it only if it is ours *)
+        end
+      | []    -> Printf.eprintf "SR (%s) has no pbds %s" (Ref.string_of sr) __LOC__; false
+      | _::_  -> true (* sr is shared, and we can see it *)
```

Comment:
Why does a single pbd signal a PBD that needs to be checked but multiple (pattren `_::_`) do not?

Reply:
AFAIK shared SRs are always visible to the master.

Reply:
yes you are right, i should check that at least one of the pbds on the shared sr belongs to the host where we are running the binary

Reply:
on reflection i am not sure if this is right.

my assumption was that we have access to a shared SR iff one of its PBDs belongs to us. this is probably not true

Reply:
after asking on slack, a shared sr should be accessible to each host in the pool, so the original code is correct. however, i will add a sanity check and print a warning if this is not the case